src/services/queue.service.ts

```diff
@@ -0,0 +1,177 @@
+import { Queue, Worker, QueueEvents, Job, QueueOptions } from 'bullmq';
+import { injectable } from 'inversify';
+import { config } from '../config';
+
+export enum JobState {
+  QUEUED = 'queued',
+  PROCESSING = 'processing',
+  COMPLETED = 'completed',
+  FAILED = 'failed',
+  CANCELLED = 'cancelled',
+  SCHEDULED = 'scheduled',
+  RETRYING = 'retrying'
+}
+
+@injectable()
+export class QueueService {
+  private queue: Queue;
+  private worker: Worker;
+  private queueEvents: QueueEvents;
+
+  constructor() {
+    const redisOptions: QueueOptions = {
+      connection: {
+        host: new URL(config.redisUrl).hostname,
+        port: Number(new URL(config.redisUrl).port) || 6379,
+        password: new URL(config.redisUrl).password || undefined
+      }
+    };
+
+    // ✅ Initialize Queue, Worker & QueueEvents
+    this.queue = new Queue(config.queueName, redisOptions);
+    this.queueEvents = new QueueEvents(config.queueName, redisOptions);
+    this.setupEventListeners();
+    this.worker = this.createWorker();
+    
+  }
+
+  public getQueueEvents() {
+    return this.queueEvents; // ✅ Exposes queue events
+  }
```

Comment:
delete redundant comments
```suggestion
  public getQueueEvents() {
    return this.queueEvents;
  }
```